Explaining syscall in release post 1.3.0 (#231)
crystal-lang/crystal-website: _releases/2022-01-06-1.3.0-released.md

## Syscalls

> We are adding an experimental API to create native system calls in Linux architectures ([#10777](https://github.com/crystal-lang/crystal/pull/10777)). This is a first step towards supporting Linux's io_uring IO interface to improve performance in kernels that support it. For example, the following code writes `"Hello!"` to `stdout` using the call generated by the macro `Syscall.def_syscall`.

Reviewer: Syscalls are not specific to Linux and should work for many other platforms eventually. Only Linux is supported for now because that's what we started with (motivated by IO uring).

```suggestion
We added an experimental API to create native system calls ([#10777](https://github.com/crystal-lang/crystal/pull/10777)). Only Linux is supported for now. This is a first step towards supporting Linux's `io_uring` interface to improve IO performance.

For example, the following code writes `"Hello!"` to `stdout` using the call generated by the macro `Syscall.def_syscall`.
```
Splitting the last paragraph because the example has nothing to do with `io_uring`. 

I don't think we need to comment on the internal behaviour of the syscall in the example. That's irrelevant for the demonstration. Maybe just copy the wording from the API docs? https://crystal-lang.org/api/master/Syscall.html#def_syscall%28name%2Creturn_type%2C%2Aargs%29-macro
> To define system calls open a module and use the `def_syscall` macro. Pass in the system call name, the return type and its arguments.

Author: > I don't think we need to comment on the internal behaviour of the syscall in the example. That's irrelevant for the demonstration.

I have no idea what this is referring to in the original text 🤔 In any case, I will simply put the link to the docs.

Reviewer: I was refering to ``writes `"Hello!"` to `stdout` ``. For this demonstration it's irrelevant *what* the syscall does, it just matters *how* to call it.

## Syscalls

We added an experimental API to create native system calls ([#10777](https://github.com/crystal-lang/crystal/pull/10777)). Only Linux is supported for now. This is a first step towards supporting Linux's `io_uring` interface to improve IO performance.

> For example, the following code writes `"Hello!"` to `stdout` using the call generated by the macro [`Syscall.def_syscall`](https://crystal-lang.org/api/1.3.0/Syscall.html#def_syscall%28name%2Creturn_type%2C%2Aargs%29-macro).

Reviewer: ```suggestion
To define system calls open a module and use the [`Syscall.def_syscall`](https://crystal-lang.org/api/1.3.0/Syscall.html#def_syscall%28name%2Creturn_type%2C%2Aargs%29-macro) macro. Pass in the system call name, the return type and its arguments. 
```

Author: Further refined:
```suggestion
To define system calls open a module and use the [`Syscall.def_syscall`](https://crystal-lang.org/api/1.3.0/Syscall.html#def_syscall%28name%2Creturn_type%2C%2Aargs%29-macro) macro. As demonstrated in the following example, you need to pass in the system call name, the return type and its arguments.
```
